src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTraceSymbols.CoreCLR.cs

```diff
@@ -191,16 +216,19 @@ private unsafe static OpenedReader TryOpenReaderForInMemoryPdb(IntPtr inMemoryPd
             return result;
         }
 
-        private static OpenedReader TryOpenReaderFromAssemblyFile(string assemblyPath, IntPtr loadedPeAddress, int loadedPeSize)
+        private static OpenedReader TryOpenReaderFromAssemblyFile(string assemblyPath, IntPtr loadedPeAddress, int loadedPeSize, Stream peStream)
         {
-            Debug.Assert(assemblyPath != null);
+            Debug.Assert(assemblyPath != null || peStream != null);
             Debug.Assert(loadedPeAddress != IntPtr.Zero);
 
-            // TODO: When/if PEReader supports reading loaded PE files use loadedPeAddress and loadedPeSize instead of opening the file again.
-            Stream peStream = TryOpenFile(assemblyPath);
             if (peStream == null)
             {
-                return null;
+                // TODO: When/if PEReader supports reading loaded PE files use loadedPeAddress and loadedPeSize instead of opening the file again.
```

Comment:
If you don't plan to do this before checking in you should remove at least the "TODO" part.